### DISCUSSION TITLE
Migrating pbkdf2 hashed passwords to bcrypt passwords.

### DISCUSSION THREAD

**Discussion by connor-hitchcock** [ory/kratos#2432] [Q&A] [answered]
From reading your documentation, source code and discussions. It appears that if you have a `pdkdf2` hashed password stored in the identity credentials table in the db, then when a user logins in, the password is hashed using `pdkdf2` and compared with the one in the database and then migrated over to either `BCrypt` or `Argon2`. 

So I have spent this afternoon checking if this functionality works as I expected it to with no avail. I did the following:

Within the source code their is a cypress test with a `pbkdf2` hashed password stored as `$pbkdf2-sha256$i=1000,l=128$e8/arsEf4cvQihdNgqj0Nw$5xQQKNTyeTHx2Ld5/JDE7A`. This format appears to be split up in `algorithm`, `sub-algorithm`, `iterations`, `length`, `salt` (base64), `hash` (base64).

I tried recreating this using the
password: `orykratos`
salt: `XPJutFhXboondLG8UxeyhS==` (base64)
sub-algorithm: `sha256`
iterations: `10000`
hash: `YZrP4NZDtuPGVdRnWWhdqg==` (base64)

On a locally hosted instance of Ory Kratos connected to a SQLite database I put the following string `$pbkdf2-sha256$i=10000,l=128$XPJutFhXboondLG8UxeyhS$YZrP4NZDtuPGVdRnWWhdqg` into the hashed password section within the identity credentials table in the db. I tried using the API Login flow to login, but have been unable to. I was wondering if I am doing something wrong and or if the functionality I'm assuming exists does not.

Cheers,
Connor.

**Comment by vinckr** [comment, marked as answer]
Hello @connor-hitchcock 
terribly sorry about the late response! Are you still facing this issue?
Are you referring to this [PDKF2 documentation](https://www.ory.sh/docs/kratos/concepts/credentials/username-email-password)?
